manifests/satellite.pp

```diff
@@ -34,11 +37,17 @@
   $_mydestination = pick($mydestination, $postfix::mydestination)
   $_mynetworks = pick($mynetworks, $postfix::mynetworks)
   $_relayhost = pick($relayhost, $postfix::relayhost)
+  $_masquerade_classes    = pick($masquerade_classes, $postfix::masquerade_classes)
+  $_masquerade_domains    = pick($masquerade_domains, $postfix::masquerade_domains)
+  $_masquerade_exceptions = pick($masquerade_exceptions, $postfix::masquerade_exceptions)
```

Comment:
`pick()` will fail if all it's arguments are `undef`.  Prefer [pick_default()](https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#pick_default-1), this should help with CI failures:

> Contrary to the pick() function, the pick_default does not fail if all arguments are empty. This allows pick_default to use an empty value as default.

Reply:
I've switched to `pick_default`, but I fear I'll still need a bit of help with the tests....